src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -1,223 +1,247 @@
+					keySelector: binding => binding.Position.ToString(),
+					elementSelector: binding => binding.ParamName
+				);
+
+				ctx.ReportDiagnostic(
+					Diagnostic.Create(
+						descriptor: Diagnostics.TooManyUnnamedArgs,
+						location: expr.GetLocation(),
+						properties: fixerContext
+					)
+				);
+
+				return;
+			}
+
+			// Literal arguments should always be named
+			foreach( var arg in unnamedArgs ) {
+				// Check if the argument type is literal
+
+				if( arg.Syntax.Expression is LiteralExpressionSyntax) {
+					var fixerContext = new Dictionary<string, string>();
+					fixerContext.Add( arg.Position.ToString(), arg.ParamName ); // Add the position and parameter name to the code-fix
+
+					ctx.ReportDiagnostic( Diagnostic.Create(
+							descriptor: Diagnostics.LiteralArgShouldBeNamed,
+							location: arg.Syntax.Expression.GetLocation(),
+							properties: fixerContext.ToImmutableDictionary(),
+							messageArgs: arg.ParamName
+						)
+					);
+				}
+			}
+
+			// TODO: if there are duplicate typed args then they should be named
+			// These will create a bit more cleanup. Fix should probably name
+			// all the args instead to avoid craziness with overloading.
+		}
+
+		/// <summary>
+		/// Get the arguments which are unnamed and not "params"
+		/// </summary>
+		private static IEnumerable<ArgParamBinding> GetUnnamedArgs(
+			SemanticModel model,
+			ArgumentListSyntax args
+		) {
+			for( var idx = 0; idx < args.Arguments.Count; idx++ ) {
+				var arg = args.Arguments[idx];
+
+				// Ignore args that already have names
+				if( arg.NameColon != null ) {
+					continue;
+				}
+
+				var param = arg.DetermineParameter(
+					model,
+
+					// Don't map params arguments. It's okay that they are
+					// unnamed. Some things like ImmutableArray.Create() could
+					// take a large number of args and we don't want anything to
+					// be named. Named params really suck so we may still
+					// encourage it but APIs that take params and many other
+					// args would suck anyway.
+					allowParams: false
+				);
+
+				// Not sure if this can happen but it'd be hard to name this
+				// param so ignore it.
+				if( param == null ) {
+					continue;
+				}
+
+				// arg.DetermineParameters() will return the first params
+				// argument, even with allowParams:false ; We should ignore
+				// any params arguments since named params is gross.
+				if( param.IsParams ) {
+
+					continue;
+				}
+
+				// IParameterSymbol.Name is documented to be possibly empty in
+				// which case it is "unnamed", so ignore it.
+				if( param.Name == "" ) {
+					continue;
+				}
+
+				// C# allows us to create variables with the same names as reserved keywords,
+				// as long as we prefix with @ (e.g. @int is a valid identifier)
+				// So any parameters which are reserved must have the @ prefix
+				string paramName;
+				SyntaxKind paramNameKind = SyntaxFacts.GetKeywordKind( param.Name );
```

Comment:
When there is no match this helper returns `SyntaxKind.None`: http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Syntax/SyntaxKindFacts.cs,960

Could we just check for that, and if so add `@`?